packages/foam-vscode/src/features/filtered-resources.ts

```diff
@@ -0,0 +1,252 @@
+import * as path from 'path';
+import * as vscode from 'vscode';
+import * as fs from 'fs';
+import {
+  IDataStore,
+  URI,
+  FoamWorkspace,
+  Resource,
+  isAttachment,
+  isPlaceholder,
+  getTitle,
+} from 'foam-core';
+import micromatch from 'micromatch';
+import {
+  FilteredResourcesConfig,
+  FilteredResourcesConfigGroupBy,
+} from '../settings';
+import { getContainsTooltip, getNoteTooltip } from '../utils';
+import { dirname, join } from 'path';
+import { FoamFeature } from '../types';
+import { commands } from 'vscode';
+
+const feature: FoamFeature = {
+  activate: (context: vscode.ExtensionContext) => {
+    context.subscriptions.push(
+      commands.registerCommand(
+        'foam-vscode.open-placeholder-note',
+        async (uri: vscode.Uri) => {
+          let dir: string;
+
+          if (vscode.workspace.workspaceFolders) {
+            dir = vscode.workspace.workspaceFolders[0].uri.fsPath.toString();
+          }
+
+          if (!dir) {
+            const activeFile = vscode.window.activeTextEditor?.document;
+            dir = activeFile ? dirname(activeFile.uri.fsPath) : null;
+          }
+
+          if (dir) {
+            const path = join(dir, `${uri.path}.md`);
+            await fs.promises.writeFile(path, `# ${uri.path}`);
+            const ur = vscode.Uri.file(path);
+            await vscode.window.showTextDocument(ur, {
+              preserveFocus: false,
+              preview: false,
+            });
+          }
+        }
+      )
+    );
+  },
+};
+
+export default feature;
+
+export class FilteredResourcesProvider
+  implements vscode.TreeDataProvider<FilteredResourceTreeItem> {
+  // prettier-ignore
+  private _onDidChangeTreeData: vscode.EventEmitter<FilteredResourceTreeItem | undefined | void> = new vscode.EventEmitter<FilteredResourceTreeItem | undefined | void>();
+  // prettier-ignore
+  readonly onDidChangeTreeData: vscode.Event<FilteredResourceTreeItem | undefined | void> = this._onDidChangeTreeData.event;
+  // prettier-ignore
+  private groupBy: FilteredResourcesConfigGroupBy = FilteredResourcesConfigGroupBy.Folder;
+  private exclude: string[] = [];
+  private filteredResources: Resource[] = [];
+  private root = vscode.workspace.workspaceFolders[0].uri.path;
+
+  constructor(
+    private workspace: FoamWorkspace,
+    private dataStore: IDataStore,
+    private filteredNoteType: string,
```

Comment:
not sure if note type really describes what this is used for. This to me is almost the ID of the provider (as e.g. it needs to be unique to work with the settings commands)

Reply:
I went with `providerId`